src/components/SearchBox.vue

```diff
@@ -1,75 +1,235 @@
+                }
+                else {
+                  item = this.filterTimeSliderCreator(item)
+                  item !== "" ? newCombinedFilter += "&fq=" + item : null;
+                }
+                })
             }
           }
-          this.updateQuery(this.searchQuery + fixedFilters)
-          //console.log(filters);
-          this.$router.push({ name: "Search", params: { query: this.searchQuery + fixedFilters, location: this.instance } });
-          e.preventDefault();
+          //Check if timefilters are applied.
+          //If not, check if they should be.
+          //If they should be, apply them at the end.
+          if(this.time === true && newCombinedFilter.includes("ff_primaryobject_year_from_i") === false && this.timeFrom !== this.searchOptions.timeOptions.min) {
+            newCombinedFilter += "&fq=ff_primaryobject_year_from_i:[" + this.timeFrom + " TO *]";
+          }
+          if(this.time === true && newCombinedFilter.includes("ff_primaryobject_year_to_i") === false && this.timeTo !== this.searchOptions.timeOptions.max) {
+            newCombinedFilter += "&fq=ff_primaryobject_year_to_i:[" + this.timeTo + " TO *]";
+          }
+          //Update the query variable in the store, and fire the search.
+          this.updateQuery(this.searchQuery + newCombinedFilter)
+          this.$router.push({ name: "Search", params: { query: this.searchQuery + newCombinedFilter, instance: this.instance } });
+          e ? e.preventDefault() : null;
+        }
+      },
+      queryTimeChanged() {
+        let proceed = false;
+        // We make sure we're on a Search instance. 
+        //if we're not we don't have to go though all this, because we'll always fire a search on the instance route no matter what.
+        if(this.time === true && this.$router.history.current.name === "Search") {
+          let filters = this.$route.params.query ? this.$route.params.query.split("fq=") : ""
+          //If we have more han one filter (that means, more than the query, we check them out)
+          filters.length > 1 ? filters.forEach(item => {
+              if(item.includes("ff_primaryobject_year_from_i")) {
+                //console.log("1",parseInt(this.$_returnTimeFromQueryString(item)), this.timeFrom)
+                //We check if the value matches the current set value in the slider. If so, we don't grant a search. Other wise, we do.
+                parseInt(this.$_returnTimeFromQueryString(item)) === this.timeFrom ? null : proceed = true;
+              }
+              if(item.includes("ff_primaryobject_year_to_i")) {
+                //console.log("3", parseInt(this.$_returnTimeFromQueryString(item)), this.timeTo)
+                //We check if the value matches the current set value in the slider. If so, we don't grant a search. Other wise, we do.
+                parseInt(this.$_returnTimeFromQueryString(item)) === this.timeTo ? null : proceed = true;
+              }
+          }) : null;
+        }
```

Comment:
This is hard to read. I know the comments take a lot of space but you have a long ternary with a loop and if statements with ternary operators inside. The ternaries inside the loop are nice and clean so my humble suggestion is to just change the first ternary to an if statement.

Reply:
Good suggestion, i'll change that.

Reply:
Actually, this is a REALLY good catch. ternary operations are excellent when you have an if else condition, but it really should be used when you NEED both conditions, no? In this canse, the else is null, which kinda makes it redundant. So a if() clause should do the trick instead and makes it far easier to read, since there's no redundant code.